internal/fwschemadata/data_set_at_path.go

```diff
@@ -28,9 +28,52 @@ import (
 // Lists can only have the next element added according to the current length.
 func (d *Data) SetAtPath(ctx context.Context, path path.Path, val interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
-
 	ctx = logging.FrameworkWithAttributePath(ctx, path.String())
 
+	if v, ok := val.(tftypes.Value); ok {
```

Comment:
I'm not sure the exact usage of this will encounter this problem, but one thing missing from this is validating that the value can be used with the actual type in the framework schema. That is logic used by custom types though so if the developer is always building the schema from a different SDK, it'll never have a custom type.

Perhaps a future thing we might run into if someone, for whatever reason, tries to use this reflection code with a real framework schema with a custom type + `tftypes.Value` (they should just be using `attr.Value` in that case)

-----

TLDR: it's probably fine that we skip validation